File: Python/Arrays/prefixSum.py
def getSum(arr, l , r):
    pSum = list()
    pSum.append(arr[0])
    for i in range(1, n):
        pSum.append(pSum[i-1] + arr[i])
    if(l == 0):
        return pSum[r]
    else:
        return pSum[r] - pSum[l-1]

if __name__ == "__main__":
    n = int(input("Enter the size of the array : "))
    arr = list()
    print("Enter the elements of the array : ")
    for i in range(0, n):
        k = int(input())
        arr.append(k)
    l = int(input("Enter the lower bound to find the sum : "))
    r = int(input("Enter the upper bound to find the sum : "))
    print(getSum(arr, l, r))

File: Java/Arrays/prefixSum.java
import java.util.*;

class prefixSum{

    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);
        int n = 0, l = 0, r = 0;
        System.out.print("Enter the size of the array : ");
        n = sc.nextInt();
        ArrayList<Integer> arr = new ArrayList<>();
        System.out.println("Enter the elements of the array : ");
        for(int i=0;i<n;i++){
            int k = sc.nextInt();
            arr.add(k);
        }
        System.out.print("Enter the lower bound to find the sum : ");
        l = sc.nextInt();
        System.out.print("Enter the higher bound to find the sum : ");
        r = sc.nextInt();
        System.out.println(findSum(arr, l, r));
    }

    public static int findSum(ArrayList<Integer> arr, int l, int r){
        ArrayList<Integer> pSum = new ArrayList<>();
        int n = arr.size();
        pSum.add(arr.get(0));
        for(int i=1;i<n;i++){
            pSum.add(pSum.get(i-1) + arr.get(i));
        }
        if(l == 0){
            return pSum.get(r);
        }
        else{
            return pSum.get(r) - pSum.get(l-1);
        }
    }
};